plugins/mcreator-core/blockly/js/field_data_list_dropdown.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * This class represents a data list field that can be used to choose value from potentially short data lists.
+ */
+class FieldDataListDropdown extends Blockly.FieldDropdown {
+
+    constructor(datalist = '', opt_validator, opt_config) {
+        super(function () {
+            return arrayToBlocklyDropDownArray(javabridge.getListOf(datalist));
+        }, opt_validator, opt_config);
+        this.type = datalist;
+        this.maxDisplayLength = 75;
+    };
+
+    // Configure the field given a map of settings
+    // Create the field from the json definition
+    static fromJson(options) {
+        return new this(Blockly.utils.parsing.replaceMessageReferences(options['datalist']), undefined, options);
+    };
+
+    // Update the value of this selector

```

Comment:
You shouldn't override `doValueUpdate_` and `getText_` for the dropdown, since values are stored in an array of `["readable name", "actual value"]` pairs. 
Also, none of the datalists provide readable names, so this part could be skipped entirely (or moved as part of another PR, along with the change to `BlocklyJavascriptBridge`)